docs/how_tos/feedback.rst

```diff
@@ -0,0 +1,337 @@
+
+The payment module's `payment/data/service.js`_ (also dubbed the
+`PaymentApiService`_ in payment/data/sagas.js) funnels many but not all API
+responses through `transformResults`_, which recursively converts snake_case
+dictionary keys of the response into camelCase.
+
+This was done because the convention for Python backends generating the API
+responses is to use snake_case, while the convention for JavaScript frontends
+processing the API responses is to use camelCase.
+
+See `payment/data/service.js`_ for if and when `transformResults`_ is used.
+
+.. _PaymentApiService: https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/payment/data/sagas.js#L34
+.. _transformResults: https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/payment/data/utils.js#L158-L172
+
+
+What about internationalization?
+--------------------------------
+
+Prefer using ``error_code``. It uses a pre-built component which should support
+internationalization like any other component with translatable strings in
+frontend-app-payment.
+
+
+For more information
+--------------------
+
+See `Feedback module's README and components`_, and especially the
+implementation in `feedback/data/sagas.js`_.
+
+.. _Feedback module's README and components: https://github.com/openedx/frontend-app-payment/tree/master/src/feedback
+.. _feedback/data/sagas.js: https://github.com/openedx/frontend-app-payment/blob/master/src/feedback/data/sagas.js
+
+
+Future work
+-----------
+
+Remove special error codes
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+There are certain error codes which receive special handling due to using
+pre-built APIs that do not follow the format of the feedback module in the:
+
+* `feedback/data/sagas.js`_
+
+    * ``basket-changed-error-message``
+    * ``transaction-declined-message``
+    * ``error_message`` in URL parameters
+
+* `payment/data/handleRequestError.js`_
+
+    * ``sku_error`` (Not an error code. See also `handleSDNCheckFailure`_.)
+    * ``payment_intent_unexpected_state``
+
+It may be beneficial for future refactors to minimize, deprecate, or
+despecialize the error codes with special handling.
+
+.. _handleSDNCheckFailure: https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/feedback/data/sagas.js#L67-L79
+
+Remove special fields names
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When using the `Dictionary of field errors`_ format, certain key names in
+the dictionary of field errors will be transformed into another field name
+using a utility function called ``convertKeyNames``. See `utils.js`_.
+
+It would be good to deprecate this transformation by having the backend send
+correct key names.
+
+.. _utils.js: https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/payment/data/utils.js#L185-L188
+
+Tame throw-catch bonanza
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+TODO
```

Comment:
i have some info on why this is done (observational; might want to chat)

Reply:
I just added a note from our conversation to the latest push to this PR.